src/raftstore/store/snap.rs

```diff
@@ -61,10 +66,120 @@ impl Display for SnapKey {
     }
 }
 
-/// Name prefix for the self-generated snapshot file.
-const SNAP_GEN_PREFIX: &'static str = "gen";
-/// Name prefix for the received snapshot file.
-const SNAP_REV_PREFIX: &'static str = "rev";
+pub enum SendSnapshotFileReader {
```

Comment:
server.snap_runner needs to hold the ownership of Writer. And RecvReader needs to expose the concrete types when applying snapshot, because the worker applies the snapshot by two different ways: the old format readers decode key and value, while the new format one lists its cf files and move them to rocksdb, it's hard to make a common trait.